Mathlib/Algebra/Module/DirectLimitAndTensorProduct.lean

```diff
@@ -97,32 +97,16 @@ noncomputable def directLimitLeft :
 -/
 noncomputable def directLimitRight :
     M ⊗[R] DirectLimit G f ≃ₗ[R] DirectLimit (M ⊗[R] G ·) (M ◁ f) :=
-  LinearEquiv.ofLinear
-    (DirectLimit.lift _ _ _ _
-      (fun i ↦ of R ι _ _ i ∘ₗ (TensorProduct.comm R M (G i)).symm.toLinearMap)
-      fun i j h x ↦ x.induction_on (by aesop) (fun m g ↦
-        show of R ι (M ⊗[R] G ·) _ j ((f i j h).lTensor M (g ⊗ₜ m)) = _ from of_f) (by aesop))
-    (DirectLimit.lift _ _ _ _
-      (fun i ↦ of R ι _ _ i ∘ₗ (TensorProduct.comm R M (G i)).toLinearMap)
-      fun i j h x ↦ x.induction_on (by aesop) (fun m g ↦
-        show of R ι (G · ⊗[R] M) _ j ((f i j h).rTensor M (g ⊗ₜ m)) = _ from of_f) (by aesop))
-    ((isEmpty_or_nonempty ι).elim (fun _ ↦ by ext; apply Subsingleton.elim)
-      fun inst_ ↦ FunLike.ext _ _ fun x ↦ x.induction_on
-        fun i g ↦ g.induction_on (by aesop) (fun g m ↦ by simp [lift_of]) <| by aesop)
-    ((isEmpty_or_nonempty ι).elim (fun _ ↦ by ext; apply Subsingleton.elim)
-      fun inst_ ↦ FunLike.ext _ _ fun x ↦ x.induction_on
-        fun i g ↦ g.induction_on (by aesop) (fun g m ↦ by simp [lift_of]) <| by aesop) ≪≫ₗ
-  directLimit f M ≪≫ₗ TensorProduct.comm _ _ _
+  TensorProduct.comm _ _ _ ≪≫ₗ directLimitLeft f M ≪≫ₗ
+  Module.DirectLimit.congr _ _ (fun i ↦ TensorProduct.comm _ _ _) fun i j h ↦ FunLike.ext _ _
+    fun x ↦ by refine x.induction_on ?_ ?_ ?_ <;> aesop
```

Comment:
You shouldn't be usign `FunLike.ext` here any more; the `ext` tactic ought to have a better one available for you that eliminates the `induction_on`.

Reply:
(I think it's `TensorProduct.ext` or `TensorProduct.ext'`)

Reply:
`show_term ext x` tells me that Lean find `refine LinearMap.ext fun x ↦ ?_ x`, we really want it to find `TensorProduct.ext` right?